docs/tutorial/t-spark-monitoring.md

```diff
@@ -0,0 +1,350 @@
+
+```bash
+# Deploy s3-integrator
+juju deploy s3-integrator -n1 --channel edge
+
+# Configure s3-integrator with correct bucket name, directory path and endpoint
+juju config s3-integrator bucket=spark-tutorial path="spark-events" endpoint=http://$S3_ENDPOINT
+
+# Sync S3 credentials with s3-integrator
+juju run s3-integrator/leader sync-s3-credentials \
+  access-key=$ACCESS_KEY secret-key=$SECRET_KEY
+
+# Integrate s3-integrator with Spark History Server
+juju integrate s3-integrator spark-history-server-k8s
+```
+
+Let's view the status of Juju model at this point with `juju status --watch 1s --relations` command. Once the charms have completed deployment and integration, the status should appear simlar to the following:
+
+
+```
+Model           Controller      Cloud/Region        Version  SLA          Timestamp
+history-server  spark-tutorial  microk8s/localhost  3.1.7    unsupported  07:29:35Z
+
+App                       Version  Status  Scale  Charm                     Channel     Rev  Address         Exposed  Message
+s3-integrator                      active      1  s3-integrator             edge         14  10.152.183.87   no       
+spark-history-server-k8s           active      1  spark-history-server-k8s  3.4/stable   15  10.152.183.111  no       
+
+Unit                         Workload  Agent  Address      Ports  Message
+s3-integrator/0*             active    idle   10.1.29.167         
+spark-history-server-k8s/0*  active    idle   10.1.29.169         
+
+Integration provider               Requirer                                 Interface            Type     Message
+s3-integrator:s3-credentials       spark-history-server-k8s:s3-credentials  s3                   regular  
+s3-integrator:s3-integrator-peers  s3-integrator:s3-integrator-peers        s3-integrator-peers  peer   
+```
+
+The Spark History Server is now successfully configured to read event logs from S3 bucket. Let's run a simple job so that Spark can generate some event logs. We're going to use the same `count_vowels.py` example we used earlier, which already exists in the `spark-tutorial` bucket.
+
+
+```bash
+spark-client.spark-submit \
+    --username spark --namespace history-server \
+    --deploy-mode cluster \
+    s3a://spark-tutorial/count_vowels.py
+```
+
+The Spark History Server comes with a Web UI for us to view and monitor the jobs submitted to Spark Cluster. The web UI can be accessed at port 18080 of the IP address of the `spark-history-server-k8s/0` unit. However, it is a good practice to access it via Ingress rather than directly accessing the unit IP address. Using an Ingress will allow us to have a common entrypoint to the apps running in the Juju model. We can add an Ingress by deploying and integrating [`traefik-k8s`](https://charmhub.io/traefik-k8s) charm with `spark-history-server-k8s`.
+
+```bash
+# Deploy traefik charm
+juju deploy traefik-k8s --channel latest/candidate --trust
```

Comment:
also here `edge`

Reply:
Using channel `latest/stable` as per the earlier comment.